astronomer/providers/google/cloud/hooks/dataproc.py

```diff
@@ -29,9 +29,20 @@ def get_cluster_client(
         :param location: (To be deprecated). The Cloud Dataproc region in which to handle the request.
         """
         client_options, region = self._get_client_options_and_region(region=region, location=location)
-        return ClusterControllerAsyncClient(
-            credentials=self._get_credentials(), client_info=CLIENT_INFO, client_options=client_options
-        )
+        try:
+            # for apache-airflow-providers-google<8.4.0
+            return ClusterControllerAsyncClient(
+                credentials=self._get_credentials(),
+                client_info=CLIENT_INFO,
+                client_options=client_options,
+            )
+        except AttributeError:  # pragma: no cover
+            # for apache-airflow-providers-google>=8.4.0
+            return ClusterControllerAsyncClient(
+                credentials=self.get_credentials(),  # type: ignore[attr-defined]
+                client_info=CLIENT_INFO,
+                client_options=client_options,
+            )
```

Comment:
why not the following?
```suggestion
        try:
            # for apache-airflow-providers-google<8.4.0
            credentials=self._get_credentials()
        except AttributeError:  # pragma: no cover
            # for apache-airflow-providers-google>=8.4.0
            credentials=self.get_credentials()  # type: ignore[attr-defined]
        return ClusterControllerAsyncClient(
            credentials=credentials,
            client_info=CLIENT_INFO,
            client_options=client_options,
        )
```

Reply:
fixed it